system/lib/libc/musl/arch/emscripten/bits/limits.h

```diff
@@ -1,12 +1 @@
-#if defined(_POSIX_SOURCE) || defined(_POSIX_C_SOURCE) \
- || defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 #define PAGE_SIZE 65536
-#define LONG_BIT 32
-#endif
-
-#if __LP64__
-#define LONG_MAX  0x7fffffffffffffffL
-#else
-#define LONG_MAX  0x7fffffffL
-#endif
-#define LLONG_MAX  0x7fffffffffffffffLL
```

Comment:
We don't need to define this in the other header? I tried to find where it was defined elsewhere and saw this:
```
$ grep -U2 -r LLONG_MAX system/lib/libc/musl/include/
system/lib/libc/musl/include/limits.h-#define LONG_MAX __LONG_MAX
system/lib/libc/musl/include/limits.h-#define ULONG_MAX (2UL*LONG_MAX+1)
system/lib/libc/musl/include/limits.h:#define LLONG_MIN (-LLONG_MAX-1)
system/lib/libc/musl/include/limits.h:#define LLONG_MAX  0x7fffffffffffffffLL
system/lib/libc/musl/include/limits.h:#define ULLONG_MAX (2ULL*LLONG_MAX+1)
system/lib/libc/musl/include/limits.h-
system/lib/libc/musl/include/limits.h-#define MB_LEN_MAX 4
```
Does that work? It uses `LLONG_MAX` before it defines it...

Reply:
otherwise lgtm

Reply:
Seems to work fine yes.   At least this is the same as upstream musl.

Reply:
Seems like this is just how the C pre-processor works, macros get a expanded at the point of use:

```
#include <stdio.h>

// BAR is not defined at the point of defining FOO
#define FOO BAR

int main() {
#define BAR 12
  printf("FOO=%d\n", FOO); // prints 12
#define BAR 13
  printf("FOO=%d\n", FOO); // prints 13
}
```

Reply:
Oh, really... very interesting. I did not know that about C macros!